test/Autofac.Test/Core/Resolving/ResolveWithGenericTypeConstraintTests.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Autofac.Test.Scenarios.GenericContraints;
+using Xunit;
+
+namespace Autofac.Test.Core.Resolving
+{
+    public class ResolveWithGenericTypeConstraintTests
```

Comment:
Could you put this test in [this fixture](https://github.com/autofac/Autofac/blob/develop/test/Autofac.Specification.Test/Registration/OpenGenericTests.cs)?

It's not obvious, but we're trying to create tests that act against the public API (the "specification tests") and keep them separate from the tests that use internals. That way as we change internals, we know the spec tests should continue to pass. We're still trying to refactor those out of the original gigantic test assembly.

Maybe create a folder/namespace under that "Graph1" to follow suit with [these complex graph resolution test stubs](https://github.com/autofac/Autofac/tree/develop/test/Autofac.Specification.Test/Resolution/Graph1) and avoid clutter in the location with the other tests.

Reply:
Should be done